File: core/src/main/java/forpleuvoir/dhwu1a/core/message/messageitem/Face.java
package forpleuvoir.dhwu1a.core.message.messageitem;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType;

import static forpleuvoir.dhwu1a.core.websocket.base.ApiKey.*;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.messageitem
 * <p>#class_name Face
 * <p>#create_time 2021/6/30 0:25
 */
public class Face extends MessageItem {

    /**
     * QQ表情编号，可选，优先高于name
     */
    @SerializedName(FACE_ID)
    public final int faceId;
    /**
     * QQ表情拼音，可选
     */
    @SerializedName(NAME)
    public final String name;

    public Face(int faceId, String name) {
        super(MessageItemType.Face);
        this.faceId = faceId;
        this.name = name;
    }

    @Override
    public String toPlainText() {
        return null;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/messageitem/Face.kt
package forpleuvoir.dhwu1a.core.message.messageitem

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.*
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.messageitem
 *
 * #class_name Face
 *
 * #create_time 2021/6/30 0:25
 */
class Face(
    /**
     * QQ表情编号，可选，优先高于name
     */
    @field:SerializedName(FACE_ID) val faceId: Int,
    /**
     * QQ表情拼音，可选
     */
    @field:SerializedName(NAME) val name: String?
) : MessageItem(MessageItemType.Face) {
    override fun toPlainText(): String {
        return String.format("/%s", name ?: faceId)
    }
}